File: src/main/groovy/com/dimaslanjaka/gradle/Utils/json/SimpleJSON.java
package com.dimaslanjaka.gradle.Utils.json;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.File;
import java.util.List;
import java.util.Map;

/**
 * Super Simple JSON Helper Java
 */
public class SimpleJSON {
	/**
	 * Convert Map to JSON and write to a file
	 */
	public static void toJson(File saveFile, Map<String, Object> stringObjectMap) {
		ObjectMapper mapper = new ObjectMapper();

		try {
			mapper.writeValue(saveFile, stringObjectMap);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	/**
	 * Convert List to JSON and write to a file
	 *
	 * @usage SimpleJSON.toJson(file, Arrays.asList ( list.toArray ()));
	 */
	public static void toJson(File saveFile, List<Object> mapList) {
		ObjectMapper mapper = new ObjectMapper();

		try {
			mapper.writeValue(saveFile, mapList);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	/**
	 * Read JSON from a file into a Map
	 */
	public static Map<String, Object> fromJson(File loadFile) {
		ObjectMapper mapper = new ObjectMapper();

		try {
			TypeReference<Map<String, Object>> type = new TypeReference<Map<String, Object>>() {
			};
			return mapper.readValue(loadFile, type);
		} catch (Exception e) {
			e.printStackTrace();
		}

		return null;
	}
}

File: src/main/kotlin/com/dimaslanjaka/gradle/utils/json/SimpleJSON.kt
package com.dimaslanjaka.gradle.utils.json

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import java.io.File

/**
 * Super Simple JSON Helper Java
 */
object SimpleJSON {
    /**
     * Convert Map to JSON and write to a file
     */
    fun toJson(saveFile: File?, stringObjectMap: MutableMap<String?, Any?>?) {
        val mapper = ObjectMapper()
        try {
            mapper.writeValue(saveFile, stringObjectMap)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    /**
     * Convert List to JSON and write to a file
     *
     * @usage SimpleJSON.toJson(file, Arrays.asList ( list.toArray ()));
     */
    fun toJson(saveFile: File?, mapList: MutableList<Any?>?) {
        val mapper = ObjectMapper()
        try {
            mapper.writeValue(saveFile, mapList)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    /**
     * Read JSON from a file into a Map
     */
    fun fromJson(loadFile: File?): MutableMap<String?, Any?>? {
        val mapper = ObjectMapper()
        try {
            val type: TypeReference<MutableMap<String?, Any?>?> = object : TypeReference<MutableMap<String?, Any?>?>() {}
            return mapper.readValue(loadFile, type)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }
}